libuserhelper.cpp

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright (c) 2020 Jolla Ltd.
+ * Copyright (c) 2020 Open Mobile Platform LLC.
+ *
+ * All rights reserved.
+ *
+ * BSD 3-Clause License, see LICENSE.
+ */
+
+#include "libuserhelper.h"
+
+#include <libuser/user.h>
+
+#include <QDebug>
+
+LibUserHelper::LibUserHelper()
+{
+}
+
+uint LibUserHelper::addGroup(const QString &group)
+{
+    struct lu_error *error = nullptr;
+    struct lu_context *context = lu_start(NULL, lu_user, NULL, NULL, NULL, NULL, &error);
+
+    if (!context) {
+        qWarning() << "Error creating context:" << lu_strerror(error);
+        lu_error_free(&error);
+        return 0;
+    }
+
+    uint rv = 0;
+    struct lu_ent *ent_group = lu_ent_new();
+    if (lu_group_default(context, group.toUtf8(), false, ent_group)) {
+        if (lu_group_add(context, ent_group, &error)) {
+            rv = lu_ent_get_first_id(ent_group, LU_GIDNUMBER);
+            if (rv == LU_VALUE_INVALID_ID) {
+                qWarning() << "Invalid group id";
+                rv = 0;
+            }
+        } else {
+            qWarning() << "Adding group failed:" << lu_strerror(error);
```

Comment:
qCWarning i.e. have logging category for this repo and component.

org.sailfishos.usermanagerd for instance

See e.g. https://github.com/sailfishos/sailfish-browser/blob/master/src/core/logging.cpp

Reply:
Logging category added

Reply:
Not seeing.

Reply:
Forgot to push to the server :)